hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java

```diff
@@ -127,6 +142,18 @@ public static TokenExpiryWrapper decodeTokenExpiry(
                 autoRenewPeriod);
     }
 
+    public static TokenExpiryWrapper decodeTokenExpiryV2(
+            @NotNull final Tuple expiryTuple, final UnaryOperator<byte[]> aliasResolver) {
+        final var second = (int) expiryTuple.get(0);
+        final var autoRenewAccount =
+                convertLeftPaddedAddressToAccountId(expiryTuple.get(1), aliasResolver);
+        final var autoRenewPeriod = (int) expiryTuple.get(2);
+        return new TokenExpiryWrapper(
+                second,
+                autoRenewAccount.getAccountNum() == 0 ? null : autoRenewAccount,
+                autoRenewPeriod);
+    }
```

Comment:
This method is almost identical to `decodeTokenExpiry()`. You can try to utilize `decodeTokenExpiry()` to handle this new cast invariant for `autoRenewPeriod` and stick to a single method.